source/extensions/filters/http/cache/key.proto

```diff
@@ -9,7 +9,13 @@ message Key {
   string path = 3;
   string query = 4;
   // True for http://, false for https://.
-  bool clear_http = 5;
+  bool clear_http = 5 [deprecated = true]; // Use scheme instead.
+  enum Scheme {
+    UNSPECIFIED = 0;
+    HTTP = 1;
+    HTTPS = 2;
+  }
+  Scheme scheme = 8;
```

Comment:
Should this have a comment?  Especially what the behavior is for unspecified, which was not previously an option.

Reply:
Done.